fraud/bad_encoding.go

```diff
@@ -144,13 +155,38 @@ func (p *BadEncodingProof) Validate(header *header.ExtendedHeader) error {
 
 	tree := wrapper.NewErasuredNamespacedMerkleTree(uint64(len(shares) / 2))
 	for i, share := range rebuiltShares {
-		tree.Push(share, rsmt2d.SquareIndex{Axis: uint(p.Index), Cell: uint(i)})
+		tree.Push(share, rsmt2d.SquareIndex{Axis: p.Index, Cell: uint(i)})
 	}
 
 	// comparing rebuilt Merkle Root of bad row/col with respective Merkle Root of row/col from block
 	if bytes.Equal(tree.Root(), root) {
-		return errors.New("invalid fraud proof: recomputed Merkle root matches the header's row/column root")
+		return errors.New(
+			"fraud: invalid fraud proof. recomputed Merkle root matches the header's row/column root",
+		)
 	}
 
 	return nil
 }
+
+// SubscribeToBEFP listens to Bad Encoding Fraud Proof and stops services once it was received
+func SubscribeToBEFP(ctx context.Context, s Subscriber, stop func(context.Context) error) {
```

Comment:
nit: What about calling this `OnBEFP` or `StopOnBEFP`? This is not really just `subscribe`ing here. But subscription until first proof + calling a stop function.

Reply:
`OnBEFP` sounds good to me.